mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -626,36 +635,45 @@ ::mediapipe::Status GlAnimationOverlayCalculator::GlSetup() {
 
     // vertex position in threespace
     attribute vec4 position;
+    attribute vec3 normal;
 
     // texture coordinate for each vertex in normalized texture space (0..1)
     attribute mediump vec4 texture_coordinate;
 
     // texture coordinate for fragment shader (will be interpolated)
-    varying mediump vec2 sample_coordinate;
+    varying mediump vec3 sample_coordinate;
+    varying mediump vec3 vNormal;
 
     void main() {
-      sample_coordinate = texture_coordinate.xy;
+      sample_coordinate = texture_coordinate.xyz;
       mat4 mvpMatrix = perspectiveMatrix * modelMatrix;
       gl_Position = mvpMatrix * position;
+
+      vec4 tmpNormal = mvpMatrix * vec4(normal, 1.0);
```

Comment:
I didn't think about this enough when I gave you this equation: mvpMatrix will transform object coordinates into our rendering coordinates.  But I don't think we actually want that for normals.  Specifically I think we want to apply rotations, but not translations.  Uniform scalings shouldn't matter, but non-uniform ones would.  So I think we need a separate matrix for applying to normals.  Assuming we're only scaling uniformly (x,y,z all the same scale-factor), then it's just rotation we'd want to use.  If I'm thinking about this right, in that case, an easy hack might be to just take "vec3 transformedZero = mvpMatrix * vec4(0.0, 0.0, 0.0, 1.0)" and then have "tmpNormal = tmpNormal - transformedZero".  Does that help improve normal results?

Reply:
I have tested this out with Lambertian shading, and it appears to perform slightly better. I am working on the lighting in a separate branch and will report back with my findings in regard to the improvements/issues when using these new normal transformations with the Lullaby shaders code.